packages/server/graphql/mutations/createPoll.ts

```diff
@@ -0,0 +1,129 @@
+const MAX_TITLE_LENGTH = 100
+
+type PollOptionsInputVariables = {
+  title: string
+}
+
+type CreatePollInputVariables = {
+  newPoll: {
+    teamId: string
+    discussionId: string
+    title: string
+    threadSortOrder: number
+    options: PollOptionsInputVariables[]
+  }
+}
+
+const createPoll = {
+  type: GraphQLNonNull(CreatePollPayload),
+  args: {
+    newPoll: {
+      type: new GraphQLNonNull(CreatePollInput),
+      description: 'The new poll including title and poll options'
+    }
+  },
+  resolve: async (
+    _source,
+    {newPoll}: CreatePollInputVariables,
+    {authToken, dataLoader, socketId: mutatorId}: GQLContext
+  ) => {
+    const viewerId = getUserId(authToken)
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+    const {teamId, discussionId, threadSortOrder, title, options} = newPoll
+
+    //AUTH
+    if (!isTeamMember(authToken, teamId)) {
+      return {error: {message: 'Not on team'}}
+    }
+
+    // VALIDATION
+    if (!title) {
+      return {error: {message: 'No poll title provided'}}
+    }
+    if (title.length > MAX_TITLE_LENGTH) {
+      return {
+        error: {
+          message: `Poll title too long, provided title is ${title.length}, max ${MAX_TITLE_LENGTH}`
+        }
+      }
+    }
+    if (options.length === 0) {
+      return {error: {message: 'No poll options provided'}}
+    }
+    if (options.length < MIN_POLL_OPTIONS) {
+      return {
+        error: {
+          message: `Poll has to have at least ${MIN_POLL_OPTIONS} options, provided ${options.length}`
+        }
+      }
+    }
+    if (options.length > MAX_POLL_OPTIONS) {
+      return {
+        error: {
+          message: `Poll can have at most ${MAX_POLL_OPTIONS} options, provided ${options.length}`
+        }
+      }
+    }
+    if (options.some((option) => !option.title)) {
+      return {error: {message: 'All options need a title'}}
+    }
+    const discussion = await dataLoader.get('discussions').load(discussionId)
+    if (!discussion) {
+      return {error: {message: 'Invalid discussion thread'}}
+    }
+    const {meetingId} = discussion
+    if (!meetingId) {
+      return {error: {message: 'Discussion does not take place in a meeting'}}
+    }
+    const meetingMemberId = MeetingMemberId.join(meetingId, viewerId)
+    const viewerMeetingMember = await dataLoader.get('meetingMembers').load(meetingMemberId)
+    if (!viewerMeetingMember) {
+      return {error: {message: 'Not a member of the meeting'}}
+    }
+
+    // RESOLUTION
+    const insertPollResult = await insertPoll({
+      createdById: viewerId,
+      teamId,
+      meetingId,
+      discussionId,
+      threadSortOrder,
+      title
+    })
+    if (insertPollResult.length === 0) {
+      return {error: {message: `Couldn't create a poll`}}
+    }
+    const [{id: pollId}] = insertPollResult
+    await insertPollOptions({pollOptions: options.map(({title}) => ({pollId, title}))})
+
+    const data = {pollId}
```

Comment:
`meetingId` is missing here but it's included in `CreatePollPayload`

Reply:
oh, good catch! I don't think it should be included there so I'm going to remove `meetingId` from `CreatePollPayload`